### DISCUSSION TITLE
Possible to capture external I2S communication?

### DISCUSSION THREAD

**Discussion by simusim** [pschatzmann/arduino-audio-tools#1084] [Q&A] [answered]
Hi Phil,

Thanks for your great libraries!

Is it possible to „read“ an working i2s communication as an „blind passenger“?
I like to build a universal spectrum analyzer with some led-panels and i am curious if it is possible to access an existing i2s datastream. For example, i‘ve build with your a2dp-library a bluetooth speaker.
Can i somehow configure a second esp32 to read the stream in parallel to the pcm5102, as a „parasite“?
When trying to learn about that i was not good enough to find out what happens to ws and bck if they are not generated by the esp32 and how to deal with them.

I know it is much easier to perform the fft on the first esp (this is working), but i like to have an universal solution. 

Does somebody have and idea for me?

Thanks a lot!
simusim

**Comment by pschatzmann** [comment, marked as answer]
I suggest to use the i2sstream as copy source and a multi output as destination
